tests/plugins/find_and_replace.test.ts

```diff
@@ -145,133 +218,200 @@ describe("basic search", () => {
-    expect(matches[3]).toStrictEqual({ col: 0, row: 4, selected: false });
-    setCellContent(model, "B1", "=1");
-    setCellContent(model, "B2", "=11");
+    expect(matches[0]).toStrictEqual({ sheetId: "s2", col: 0, row: 1, selected: true });
+    searchOptions.searchScope = "allSheets";
+    model.dispatch("UPDATE_SEARCH", { toSearch: "1", searchOptions });
     matches = model.getters.getSearchMatches();
     matchIndex = model.getters.getCurrentSelectedMatchIndex();
-    expect(matches).toHaveLength(6);
+    expect(getActivePosition(model)).toBe("A2");
+    expect(matches).toHaveLength(4);
+    expect(matchIndex).toStrictEqual(3);
+    expect(matches[3]).toStrictEqual({ sheetId: "s2", col: 0, row: 1, selected: true });
+  });
+
+  test("update cell will also update the search", () => {
+    activateSheet(model, "s2");
+    model.dispatch("UPDATE_SEARCH", { toSearch: "1", searchOptions });
+    let matches = model.getters.getSearchMatches();
+    let matchIndex = model.getters.getCurrentSelectedMatchIndex();
+    expect(matches).toHaveLength(1);
+    expect(matchIndex).toStrictEqual(0);
+    setCellContent(model, "B2", "1");
+    matches = model.getters.getSearchMatches();
+    matchIndex = model.getters.getCurrentSelectedMatchIndex();
+    expect(matches).toHaveLength(2);
     expect(matchIndex).toStrictEqual(0);
-    expect(matches[0]).toStrictEqual({ col: 1, row: 0, selected: false });
-    expect(matches[1]).toStrictEqual({ col: 0, row: 1, selected: true });
-    expect(matches[2]).toStrictEqual({ col: 1, row: 1, selected: false });
-    expect(matches[3]).toStrictEqual({ col: 0, row: 2, selected: false });
-    expect(matches[4]).toStrictEqual({ col: 0, row: 3, selected: false });
-    expect(matches[5]).toStrictEqual({ col: 0, row: 4, selected: false });
   });
 
-  test("new search when changing sheet", () => {
-    const sheet1 = model.getters.getActiveSheetId();
-    const sheet2 = "42";
-    createSheet(model, { activate: true, sheetId: sheet2 });
-    setCellContent(model, "B1", "hello");
-    setCellContent(model, "B2", "Hello");
-    setCellContent(model, "B3", "hello1");
-    activateSheet(model, sheet1);
-    model.dispatch("UPDATE_SEARCH", { toSearch: "hello", searchOptions });
+  test("update cell with formula will also update the search", () => {
+    activateSheet(model, "s2");
+    model.dispatch("UPDATE_SEARCH", { toSearch: "hey", searchOptions });
     let matches = model.getters.getSearchMatches();
     let matchIndex = model.getters.getCurrentSelectedMatchIndex();
     expect(matches).toHaveLength(2);
     expect(matchIndex).toStrictEqual(0);
-    activateSheet(model, sheet2);
+    setCellContent(model, "B2", '="hey"');
     matches = model.getters.getSearchMatches();
     matchIndex = model.getters.getCurrentSelectedMatchIndex();
     expect(matches).toHaveLength(3);
     expect(matchIndex).toStrictEqual(0);
-    activateSheet(model, sheet1);
+  });
+
+  test.skip("Will search a modified cell", () => {
```

Comment:
Skip ? And how is it the forth freaking test with a **_slightly different name that means exactly the same thing_**